lib/SIL/IR/SILFunction.cpp

```diff
@@ -1002,12 +1001,11 @@ bool SILFunction::shouldBePreservedForDebugger() const {
   if (getModule().getASTContext().LangOpts.hasFeature(Feature::Embedded))
     return false;
 
-  if (const IRGenOptions *options = getModule().getIRGenOptionsOrNull()) {
-    if (options->WitnessMethodElimination ||
-        options->VirtualFunctionElimination ||
-        options->LLVMLTOKind != IRGenLLVMLTOKind::None) {
-      return false;
-    }
+  const auto &silOptions = getModule().getOptions();
```

Comment:
Why not just add a `bool ShouldBePreservedByDebugger` to SILOptions? Then you don’t have to much a bunch of stuff around.